src/zero_copy/input.rs

```diff
@@ -10,6 +10,8 @@ pub trait Input {
     fn next(&self, offset: Self::Offset) -> (Self::Offset, Option<Self::Token>);
 
     fn span(&self, range: Range<Self::Offset>) -> Self::Span;
+
+    fn select_greater<T>(&self, left: (Self::Offset, T), right: (Self::Offset, T)) -> (Self::Offset, T);
```

Comment:
I think that's a good point, but then the question becomes how do we tell who got the furthest? Do we use 'most errors' prioritization via marker, or some other method? This solution was based off the idea that all current implementations, offsets will be monotonic. I'm not sure `zero_copy` practically allows a user to have non-monotonic offsets like you describe, at least not in a friendly way - Input is implemented for `[T]` and friends.

Reply:
> how do we tell who got the furthest?

Previously, I used the stream's iterator count to determine the furthest error. Obviously that becomes a little bit more non-trivial now that inputs are usually slices. Increasingly I'm thinking that `Span` should be an associated type on `Error` instead of `Input` so that we don't tie things like `str` to any one specific span type. I think I'm going to need to do some thinking about this an revisit it because it's not clear to me what the best thing to do is yet.